samples: nrf9160: MQTT Multi Service Sample: Fix PGPS & feature disablement (#7420)
nrfconnect/sdk-nrf: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

The devicetree overlay file is automatically applied during compilation whenever the ``thingy91_nrf9160_ns`` target is selected.

> The required Kconfig options are implicitly enabled by :ref:`CONFIG_TEMP_DATA_USE_SENSOR <CONFIG_TEMP_DATA_USE_SENSOR>`.

Reviewer: I think docs folks usually recommend, instead of saying:
```
enabled by :ref:`CONFIG_FOO`
```

saying:
```
enabled by the :ref:`CONFIG_FOO` option
```

Location tracking
=================

All matters concerning location tracking are handled in the :file:`src/location_tracking.c` file.
For the most part, this amounts to setting up a periodic location request, and then passing the results to a callback configured by the :file:`src/application.c` file.
Additionally, the following two tasks are performed:

* The :ref:`lib_location` library requires GPS assistance data (either A-GPS, P-GPS, or both) to have the fastest possible `time-to-first-fix <TTFF_>`_.
  The library requests this data automatically, but has no way to automatically receive the response.
  Therefore, all MQTT messages received from `nRF Cloud`_ must be be passed directly through to the :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps` libraries to be checked for A-GPS and P-GPS data, respectively.
  The :c:func:`location_assistance_data_handler` function handles this task, and MQTT messages are sent directly to it from the :file:`src/connection.c` file.
* The GNSS module cannot obtain a fix without the antenna first being properly configured.
  This configuration is performed by the :c:func:`gnss_antenna_configure` function.

> Note that for location readings to be visible in the nRF Cloud portal, they must be marked as enabled in the `Device Shadow <nRF Cloud Device Shadows_>`_.
> This is performed by the :file:`src/connection.c` file.

Reviewer 2: ```suggestion
.. note::
   For location readings to be visible in the nRF Cloud portal, they must be marked as enabled in the `Device Shadow <nRF Cloud Device Shadows_>`_.
   This is performed by the :file:`src/connection.c` file.
```

Author: I hesitate to put this in a proper note box, since there is no direct action required by the user under any circumstance. It is more meant as a theoretical footnote. Or would that still qualify for inclusion in a note box?

*************
|config|

Disabling key features
======================

> 

Reviewer: Please add a note that makes it clear that MQTT is only recommended for applications which need to stay connected constantly and that need to transfer data frequently.  So, while individual services in this sample can be disabled, and the rate at which active services report data can be changed, the reader should not take this as a recommendation to use MQTT inappropriately.

The following key features of this sample may be independently disabled:

* GNSS-based location tracking
* Cellular-based location tracking
* Temperature tracking
* GNSS assistance (A-GPS)
* Predictive GNSS assistance (P-GPS)

> To disable GNSS-based location tracking, set ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` to disabled.

Reviewer: See my other comment about following the CONFIG with "option"

To disable cellular-based location tracking, set ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` to disabled.

> Disabling both simultaneously will completely disable location tracking.

Reviewer 3: ```suggestion
Disabling both will completely disable location tracking.
```

*************
|config|

Disabling key features
======================

The following key features of this sample may be independently disabled:

> * GNSS-based location tracking

Reviewer 2: ```suggestion
* GNSS-based location tracking - set the ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` option to disabled.
```

> * Cellular-based location tracking

Reviewer 2: ```suggestion
* Cellular-based location tracking - set the ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` option to disabled.
```

> * Temperature tracking

Reviewer 2: ```suggestion
* Temperature tracking - set the ref:`CONFIG_TEMP_TRACKING <CONFIG_TEMP_TRACKING>` option to disabled.
```

> * GNSS assistance (A-GPS)

Reviewer 2: ```suggestion
* GNSS assistance (A-GPS) - set the :kconfig:option:`CONFIG_NRF_CLOUD_AGPS` option to disabled.
```

> * Predictive GNSS assistance (P-GPS)

Reviewer 2: ```suggestion
* Predictive GNSS assistance (P-GPS) - set the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS` option to disabled.
```

> To disable GNSS-based location tracking, set the ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` option to disabled.
> 
> To disable cellular-based location tracking, set the ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` option to disabled.

Reviewer 2: ```suggestion
```

> Disabling both options will completely disable location tracking.

Reviewer 2: ```suggestion
If you set both GNSS and cellular-based disabled, location tracking is completely disabled.
```

> To disable temperature tracking, set the ref:`CONFIG_TEMP_TRACKING <CONFIG_TEMP_TRACKING>` option to disabled.
> 
> To disable A-GPS, set the :kconfig:option:`CONFIG_NRF_CLOUD_AGPS` option to disabled.
> 
> To disable P-GPS, set the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS` option to disabled.

Reviewer 2: ```suggestion
```

> note::

Reviewer 2: ```suggestion
.. note::
```

  MQTT should only be used with applications that need to stay connected constantly or transfer data frequently.
  While this sample does allow its core features to be slowed or completely disabled, in real-world applications, you should carefully consider your data throughput and whether MQTT is an appropriate solution.

>   If you find yourself wanting to disable or excessively slow any of these features for a real-world application, other solutions, such as the `nRF Cloud Rest API`_, may be more appropriate.

Reviewer: This is good.  Small nit is that it should say `excessively slow all of these features` -- if any of them are still frequent, that's sufficient to justify MQTT.

>   If you find yourself wanting to disable or excessively slow all of these features for a real-world application, other solutions, such as the `nRF Cloud Rest API`_, may be more appropriate.

Reviewer 2: ```suggestion
  If you want to disable or excessively slow all of these features for a real-world application, other solutions, such as the `nRF Cloud Rest API`_, may be more appropriate.
```

Useful debugging options
========================

> To see all debug output for this sample, enable the ref:`CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG <CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG>` option.

Reviewer 2: ```suggestion
To see all debug output for this sample, enable the :ref:`CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG <CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL_DBG>` option.
```

To monitor the GNSS module (for instance, to see whether A-GPS or P-GPS assistance data is being consumed), enable the :kconfig:option:`CONFIG_NRF_CLOUD_GPS_LOG_LEVEL_DBG` option.

> See also the `test counter <nrf_cloud_mqtt_multi_service_test_counter_>`_.

Reviewer 2: ```suggestion
See also :ref:`nrf_cloud_mqtt_multi_service_test_counter`.
```

> Sample configuration options
> ============================

Reviewer 2: ```suggestion
Configuration options
=====================
```

   Sets the GNSS fix timeout in seconds.
   On each location sample, try for this long to achieve a GNSS fix before falling back to cellular positioning.

.. _CONFIG_LOCATION_TRACKING_SAMPLE_INTERVAL_SECONDS:

CONFIG_LOCATION_TRACKING_SAMPLE_INTERVAL_SECONDS - Location sampling interval (seconds)
   Sets the location sampling interval in seconds.

.. _CONFIG_LOCATION_TRACKING_GNSS:

CONFIG_LOCATION_TRACKING_GNSS - GNSS location tracking
   Sets whether to use GNSS location tracking.

>    Disable both this and :ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` to comletely disable location tracking.

Reviewer 3: ```suggestion
   Disable both this and :ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` to completely disable location tracking.
```

   Defaults to enabled.

.. _CONFIG_LOCATION_TRACKING_CELLULAR:

CONFIG_LOCATION_TRACKING_CELLULAR - Cellular location tracking
   Sets whether to use cellular location tracking.

>    Disable both this and :ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` to comletely disable location tracking.

Reviewer 3: ```suggestion
   Disable both this and :ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` to completely disable location tracking.
```

   Sets the GNSS fix timeout in seconds.
   On each location sample, try for this long to achieve a GNSS fix before falling back to cellular positioning.

.. _CONFIG_LOCATION_TRACKING_SAMPLE_INTERVAL_SECONDS:

CONFIG_LOCATION_TRACKING_SAMPLE_INTERVAL_SECONDS - Location sampling interval (seconds)
   Sets the location sampling interval in seconds.

.. _CONFIG_LOCATION_TRACKING_GNSS:

CONFIG_LOCATION_TRACKING_GNSS - GNSS location tracking

>    Sets whether to use GNSS location tracking.

Reviewer 2: ```suggestion
   Enables GNSS location tracking.
```

   Disable both this and :ref:`CONFIG_LOCATION_TRACKING_CELLULAR <CONFIG_LOCATION_TRACKING_CELLULAR>` to completely disable location tracking.
   Defaults to enabled.

.. _CONFIG_LOCATION_TRACKING_CELLULAR:

CONFIG_LOCATION_TRACKING_CELLULAR - Cellular location tracking

>    Sets whether to use cellular location tracking.

Reviewer 2: ```suggestion
   Enables cellular location tracking.
```

   Disable both this and :ref:`CONFIG_LOCATION_TRACKING_GNSS <CONFIG_LOCATION_TRACKING_GNSS>` to completely disable location tracking.
   Defaults to enabled.

.. _CONFIG_TEMP_DATA_USE_SENSOR:

CONFIG_TEMP_DATA_USE_SENSOR - Use genuine temperature data
   Sets whether to take genuine temperature measurements from a connected BME680 sensor, or just simulate sensor data.

.. _CONFIG_TEMP_TRACKING:

CONFIG_TEMP_TRACKING - Track temperature data

>    Sets whether to track temperature data, and report it to `nRF Cloud`_.

Reviewer 2: ```suggestion
   Enables tracking of temperature data, and reporting it to `nRF Cloud`_.
```
